homeassistant/components/nuki/lock.py

```diff
@@ -109,23 +123,40 @@ def device_state_attributes(self):
         """Return the device specific state attributes."""
         data = {
             ATTR_BATTERY_CRITICAL: self._battery_critical,
-            ATTR_NUKI_ID: self._nuki_lock.nuki_id}
+            ATTR_NUKI_ID: self._nuki_lock.nuki_id,
+            ATTR_LOCK_REACHABLE: self._lock_reachable}
         return data
 
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_OPEN
+
     def update(self):
         """Update the nuki lock properties."""
-        self._nuki_lock.update(aggressive=False)
-        self._name = self._nuki_lock.name
-        self._locked = self._nuki_lock.is_locked
-        self._battery_critical = self._nuki_lock.battery_critical
+        try:
+            self._nuki_lock.update(aggressive=False)
+        except requests.exceptions.RequestException:
+            self._lock_reachable = False
+        else:
+            self._name = self._nuki_lock.name
+            self._locked = self._nuki_lock.is_locked
+            self._battery_critical = self._nuki_lock.battery_critical
 
     def lock(self, **kwargs):
         """Lock the device."""
         self._nuki_lock.lock()
 
     def unlock(self, **kwargs):
         """Unlock the device."""
-        self._nuki_lock.unlock()
+        if self._unlatch_to_unlock:
+            self._nuki_lock.unlatch()
+        else:
+            self._nuki_lock.unlock()
+
+    def open(self, **kwargs):
```

Comment:
This method is not used?

Reply:
Which method? `Open`?

Reply:
This is overwriting this base class entity method:
https://github.com/home-assistant/home-assistant/blob/d22c3f13b2262a2a8fa4d4ef02ebfc709c3a5620/homeassistant/components/lock/__init__.py#L127

It opens the door. Some iot locks support opening the door besides locking / unlocking.

Reply:
ooh. In that case, we should not have an option to have unlock be unlatch, as `open` is meant for unlatch.

Reply:
With the Open base class method overwritten, I tried to open the lock using Siri. The command "Hey Siri, open the lock" it just unlocks, but not unlatchs. And also, in Home app, the lock is shown as a two-state button: Open and closed. But Open just unlocks. 
I suppose the right solution should be the lock to act as three-state: Open, Closed and Unlacthed. But I don't know if this depends on HA or Apple. 
Another option could be that if the Open method is implemented, use that instead of unlock.

Reply:
So we are not going to change this integration because Siri is not interacting with the lock right. Instead, you need to fix Siri. Please remove the option to treat unlock as unlatch from this PR

Reply:
Ok, I'll revert it. Where is the right place to suggest to use the Open service instead of the Unlock service in Homekit locks where available? I'm not so confident to propose a PR.

Reply:
That would be the `homekit` integration in Home Assistant.